sklearn/kmeans.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2017-2019 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from __future__ import print_function
+import numpy as np
+import os
+import timeit
+from numpy.random import rand
+from sklearn.cluster import KMeans
+from args import getArguments,coreString
+import sklearn
+import bench
+
+import argparse
+argParser = argparse.ArgumentParser(prog="kmeans.py",
+                                    description="sklearn kmeans benchmark",
+                                    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+argParser.add_argument('-x', '--filex', '--fileX', '--input',
+                       type=str, help='Points to cluster')
+argParser.add_argument('-i', '--filei', '--fileI', '--init',
+                       type=str, help='Initial clusters')
+# argParser.add_argument('-t', '--filet', '--fileT', '--tol',
+#                        type=str, help='Absolute threshold')
+argParser.add_argument('-m', '--data-multiplier', default=100,
+                       type=int, help='Data multiplier')
+args = getArguments(argParser)
+REP = args.iteration if args.iteration != '?' else 10
+num_threads, daal_version = bench.prepare_benchmark(args)
+
+
+def st_time(func):
+    def st_func(*args, **keyArgs):
+        times = []
+        for n in range(REP):
+            t1 = timeit.default_timer()
+            r = func(*args, **keyArgs)
+            t2 = timeit.default_timer()
+            times.append(t2-t1)
+            if sum(times) > 60:
+                break
+        print (min(times), end='')
+        return r
+    return st_func
+
+problem_sizes = [(args.size[0], args.size[1])]
+
+X = np.load(args.filex)
+X_init = np.load(args.filei)
+X_mult = np.vstack((X,) * args.data_multiplier)
+
+kmeans = KMeans(n_clusters=10, n_jobs=int(args.num_threads), tol=1e-16,
```

Comment:
`n_jobs` is not controlling the number of threads used by DAAL, if we are running this in IDP. What does?

Reply:
on line 30, the following code 

```python
num_threads, daal_version = bench.prepare_benchmark(args)
```

runs both `daal4py.daalinit(nthreads=args.num_threads)` and disables finiteness checking for sklearn.

Do we want to remove the n_jobs argument from `KMeans`?

Reply:
We can keep it, to ensure that upstream KMeans is run in parallel as well.